dotCMS/src/integration-test/java/com/dotcms/content/elasticsearch/util/ESMappingUtilHelperTest.java

```diff
@@ -0,0 +1,349 @@
+import com.dotmarketing.exception.DotSecurityException;
+import com.dotmarketing.portlets.structure.model.Relationship;
+import com.dotmarketing.util.Logger;
+import com.dotmarketing.util.UtilMethods;
+import com.dotmarketing.util.WebKeys.Relationship.RELATIONSHIP_CARDINALITY;
+import com.dotmarketing.util.json.JSONObject;
+import com.liferay.portal.model.User;
+import com.liferay.util.StringPool;
+import com.tngtech.java.junit.dataprovider.DataProvider;
+import com.tngtech.java.junit.dataprovider.DataProviderRunner;
+import com.tngtech.java.junit.dataprovider.UseDataProvider;
+import com.tngtech.junit.dataprovider.format.DataProviderTestNameFormatter;
+import java.io.IOException;
+import java.lang.reflect.Method;
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * Class used to validate ES mapping for fields is applied correctly considering fields types and dynamic mapping defined in `es-content-mapping.json`
+ * @author nollymar
+ */
+@RunWith(DataProviderRunner.class)
+public class ESMappingUtilHelperTest {
+
+    private static User user = APILocator.systemUser();
+    private static ContentletIndexAPI contentletIndexAPI = APILocator.getContentletIndexAPI();
+    private static ContentTypeAPI contentTypeAPI = APILocator.getContentTypeAPI(user);
+    private static ESMappingAPIImpl esMappingAPI = new ESMappingAPIImpl();
+    private static FieldAPI fieldAPI = APILocator.getContentTypeFieldAPI();
+    private static RelationshipAPI relationshipAPI = APILocator.getRelationshipAPI();
+
+    public static class PlusTestNameFormatter implements DataProviderTestNameFormatter {
+        @Override
+        public String format(Method testMethod, int invocationIndex, List<Object> arguments) {
+            return String.format("Test Name: %s. Mapping Name: %s", testMethod.getName(), arguments.get(0));
+        }
+    }
+
+    /*Data provider used to verify that the ES mapping is applied correctly and considers exclusions in the `es-content-mapping` file
+        For each scenario we define:
+            1) testCase name
+            2) Field(s) type
+            3) Field(s) variable name(s)
+            4) Expected ES mapping
+     */
+    @DataProvider(formatter = PlusTestNameFormatter.class)
+    public static Object[][] dataProviderAddMappingForFields() {
+        return new Object[][] {
+                {  "strings_as_dates", DateField.class, DataTypes.DATE,
+                        new String[] {"originalstartdate", "recurrencestart", "recurrenceend"},  "date" },
+
+                {  "strings_as_date_times", DateTimeField.class, DataTypes.DATE,
+                        new String[] {"originalstartdate", "recurrencestart", "recurrenceend"},  "date" },
+
+                {  "dates_as_text", TextField.class, DataTypes.TEXT,
+                        new String[] {"originalstartdate", "recurrencestart", "recurrenceend"},  "text" },
```

Comment:
Codacy found an issue: [The String literal "text" appears 4 times in this file; the first occurrence is on line 90](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5739222)